netann/node_signer.go

```diff
@@ -10,24 +10,30 @@ import (
 	"github.com/lightningnetwork/lnd/lnwallet"
 )
 
-// NodeSigner is an implementation of the MessageSigner interface backed by the
-// identity private key of running lnd node.
-type NodeSigner struct {
+type NodeSigner interface {
+	lnwallet.MessageSigner
+	SignDigestCompact(digest []byte) ([]byte, error)
```

Comment:
hmmm... I wonder why they don't implement the full `SingleKeyDigestSigner` interface

Reply:
Heh, because:
```
type SingleKeyDigestSigner interface {
	SignDigest(digest [32]byte) (*btcec.Signature, error)
	SignDigestCompact(digest [32]byte) ([]byte, error)
}
```
but:
```
func (n *NodeSignerImpl) SignCompact(msg []byte) ([]byte, error)
func (n *NodeSignerImpl) SignDigestCompact(hash []byte) ([]byte, error)
```
I can't tell what portion of this is on purpose ..

Reply:
ug